src/runtime/runtime.ts

```diff
@@ -221,12 +253,81 @@ export class Runtime {
     // we could eliminate it if the Manifest object takes care of this.
     const id = `in-memory-${Math.floor((Math.random()+1)*1e6)}.manifest`;
     // TODO(sjmiles): this is a virtual manifest, the fileName is invented
-    const localOptions = {id, fileName: `./${id}`, loader};
-    return Manifest.parse(content, {...localOptions, ...options});
+    const opts = {id, fileName: `./${id}`, loader, ...options};
+    return Manifest.parse(content, opts);
+  }
+
+  async parseFile(path: string, options?): Promise<Manifest> {
+    const content = await this.loader.loadResource(path);
+    const opts = {id: path, fileName: path, loader: this.loader, ...options};
+    return this.parse(content, opts);
+  }
+
+  async resolveRecipe(arc: Arc, recipe: Recipe): Promise<Recipe | null> {
+    if (this.normalize(recipe)) {
+      if (recipe.isResolved()) {
+        return recipe;
+      }
+      const resolver = new RecipeResolver(arc);
+      const plan = await resolver.resolve(recipe);
+      if (plan && plan.isResolved()) {
+        return plan;
+      }
+      warn('failed to resolve:\n', (plan || recipe).toString({showUnresolved: true}));
+    }
+    return null;
   }
 
+  normalize(recipe: Recipe): boolean {
+    if (Runtime.isNormalized(recipe)) {
+      return true;
+    }
+    const errors = new Map();
+    if (recipe.normalize({errors})) {
+      return true;
+    }
+    warn('failed to normalize:\n', errors, recipe.toString());
+    return false;
+  }
+
+  static isNormalized(recipe: Recipe): boolean {
+    return Object.isFrozen(recipe);
+  }
+
+  // TODO(sjmiles): redundant vs. newArc, but has some impedance mismatch
+  // strategy is to merge first, unify second
+  async spawnArc({id, serialization, context, composer, storage, portFactories}: SpawnArgs): Promise<Arc> {
+    const params = {
+      id: IdGenerator.newSession().newArcId(id),
+      fileName: './serialized.manifest',
+      serialization,
+      context,
+      storageKey: storage || 'volatile',
+      slotComposer: composer,
+      pecFactories: [this.pecFactory, ...(portFactories || [])],
+      loader: this.loader,
+      // TODO(sjmiles): maybe doesn't belong here, but empirically it's
+      // wanted in (almost?) all Arc instantiations
+      inspectorFactory: devtoolsArcInspectorFactory
+    };
+    return serialization ? Arc.deserialize(params) : new Arc(params);
+  }
+
+  // static interface for the default runtime environment
+
   static async parse(content: string, options?): Promise<Manifest> {
     return this.getRuntime().parse(content, options);
   }
 
+  static async parseFile(path: string, options?): Promise<Manifest> {
+    return this.getRuntime().parseFile(path, options);
+  }
+
+  static async resolveRecipe(arc: Arc, recipe: Recipe): Promise<Recipe | null> {
+    return this.getRuntime().resolveRecipe(arc, recipe);
+  }
+
+  static async spawnArc(args: SpawnArgs): Promise<Arc> {
```

Comment:
Up to you, but I think it would be clearer not to provide these static helper methods, and just require callers to invoke these methods directly on an instance of Runtime. Providing all of these as static methods suggests that their functionality is some sort of global phenomenon (e.g. Runtime.spawnArc() implies that a new arc appears, whereas runtime.spawnArc() more clearly implies that you're spawning an arc inside the given runtime).

Reply:
This functionality is based on the concept that most (empirically today, all) applications have only one `runtime` object. Access to static methods on the default runtime saves plumbing a ton of `runtime`  arguments (I tried to document this in comments).

One can create multiple runtime objects and customize them, but again, empirically,  the static provide big ergonomic benefit.

Reply:
I tend to try to avoid global state, and prefer things like dependency injection. There's only one runtime instance, until there isn't, and then you have to go through and fix up all the places where that was assumed.

To me, explicitly calling `getInstance()` really calls out that you're accessing the global instance. And calling a static method like `Runtime.parse()` suggests that it doesn't have any side-effects and doesn't use anything from the Runtime instance. So it's a bit surprising to find that what looks like a fully static method is actually calling a method on a global instance.

Those are just my thoughts. If you think the improves ergonomics of static methods outweighs the increase in clarity over instance methods, then feel free to carry on :)

Reply:
I'm not quite ready to let this go, but I'm also not surprised it's controversial. 

https://github.com/PolymerLabs/arcs/issues/3999

Since the `runtime` mono-state (maybe I'm splitting hairs, but I don't consider that the same as a global) was implemented by somebody not-me, I thought maybe this had already been considered.

Reply:
... and thanks for all the review comments!